File: solutions/Javascript/Pascal's Triangle II.js
var getRow = function (rowIndex) {
  triangle = [[1]];
  while (triangle.length <= rowIndex)
    triangle.push(makeRow(triangle[triangle.length - 1]));
  return triangle[rowIndex];
};

function makeRow(row) {
  let res = [],
    i = 0,
    j = row.length;
  console.log(row);
  while (i <= j) {
    if (i === 0) res.push(row[i]);
    else if (i <= j - 1) res.push(row[i - 1] + row[i]);
    else res.push(row[j - 1]);
    i = i + 1;
  }
  return res;
}

File: solutions/cpp/Pascal's Triangle II.cpp
#include <vector>
class Solution
{
public:
    std::vector<int> makeRow(std::vector<int> row)
    {
        int i = 0, j = row.size();
        std::vector<int> res = {};
        while (i <= j)
        {
            if (i == 0)
                res.push_back(row[i]);
            else if (i <= j - 1)
                res.push_back(row[i - 1] + row[i]);
            else
                res.push_back(row[j - 1]);
            i = i + 1;
        }
        return res;
    }
    std::vector<int> getRow(int rowIndex)
    {
        std::vector<std::vector<int>> triangle = {{1}};
        while (triangle.size() <= rowIndex)
            triangle.push_back(makeRow(triangle[triangle.size() - 1]));
        return triangle[rowIndex];
    }
};